tables/tabulate-molecular-alterations.R

```diff
@@ -0,0 +1,438 @@
+    dplyr::mutate(Tumor_Sample_Barcode =  sample_id) %>%
+    dplyr::rename(Variant_Classification = status,
+                  Hugo_Symbol = gene_symbol) %>%
+    dplyr::select(Hugo_Symbol, Tumor_Sample_Barcode, Variant_Classification) %>%
+    # mutate loss and amplification to Del and Amp to fit Maftools format
+    dplyr::mutate(Variant_Classification = dplyr::case_when(Variant_Classification == "deep deletion" ~ "Del",
+                                                            Variant_Classification == "amplification" ~ "Amp",
+                                                            TRUE ~ as.character(Variant_Classification))) %>%
+    # only keep Del and Amp calls
+    dplyr::filter(Variant_Classification %in% c("Del", "Amp"))
+  
+}
+
+
+#### Read in data ------------------------------------
+histologies_df <- readr::read_tsv(metadata_file, guess_max = 10000) %>%
+  dplyr::inner_join(
+    readr::read_tsv(pal_file) %>%
+      dplyr::select(cancer_group, cancer_group_display, broad_histology, broad_histology_display)
+  )
+
+# Only keep required maf columns 
+keep_cols <-c("Hugo_Symbol", 
+              "Chromosome",
+              "Start_Position",
+              "End_Position",
+              "Reference_Allele",
+              "Tumor_Seq_Allele2",
+              "Variant_Classification",
+              "Variant_Type",
+              "Tumor_Sample_Barcode",
+              "HGVSp_Short")
+
+maf_df <- readr::read_tsv(maf_file) %>%
+  dplyr::select(keep_cols)
+
+hotspots_maf_df <- readr::read_tsv(hotspots_maf_file) %>%
+  dplyr::select(keep_cols)
+  
+# merge hotspots maf to input maf
+maf_df <- maf_df %>%
+  dplyr::bind_rows(hotspots_maf_df) %>%
+  dplyr::distinct()
+
+
+# read fusion data
+fusion_df <- readr::read_tsv(fusion_file)
+
+
+# read and join cnv with histology information
+cnv_autosomes_df <- readr::read_tsv(cnv_autosomes_file) %>%
+  dplyr::left_join(
+    dplyr::select(histologies_df, 
+                  Kids_First_Biospecimen_ID,
+                  germline_sex_estimate),
+            by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")
+  )
+
+# combine autosomes with x/y cnv
+cnv_df <- dplyr::bind_rows(
+  cnv_autosomes_df, 
+  readr::read_tsv(cnv_xy_file)
+)
+
+
+## Filter to relevant tumors --------------------------------------
+
+# The final output should only tabuluate alterations for N=1074 tumors 
+# We'll need to identify those samples, including their sample_id and biospecimen.
+
+# In addition, we'll note if a sample_id is ambiguous. Ambiguous sample_id's will 
+# have more than 2 rows associated with it in the histologies file when looking at 
+# tumor samples -- that means we won't necessarily be able to determine **when an 
+# WGS/WXS assay maps to an RNA-seq assay** for the purpose of tabulating molecular alterations
+
+tumor_sample_ids_df <- histologies_df %>%
+  dplyr::filter(sample_type == "Tumor",
+                composition == "Solid Tissue") %>%
```

Comment:
We want to include cell lines, too, per the response to reviewers and https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1692#issue-1626351515 (adding emphasis here):

> A CSV file that contains a table of molecular alterations included in Figure 2 for all tumor **and cell line samples** in the PBTA dataset

Reply:
I expect this also impacts your use of `openpbta_n_tumors` throughout. You could add `openpbta_n_cell_line` and then have the check be for the length of `openpbta_n_tumors + openpbta_n_cell_line`.